pkg/server/registry/apigroups/acorn/events/strategy.go

```diff
@@ -182,14 +223,57 @@ func stripQuery(opts storage.ListOptions) (q query, stripped storage.ListOptions
 	return
 }
 
+func parseTimeBound(raw string, now internalv1.MicroTime, lower bool) (*internalv1.MicroTime, error) {
+	// Try to parse raw as a duration string
+	var errs []error
+	duration, err := time.ParseDuration(raw)
+	if err == nil {
+		if lower {
+			duration *= -1
+		}
+
+		return z.P(internalv1.NewMicroTime(now.Add(duration))), nil
+	}
+	errs = append(errs, fmt.Errorf("%s is not a valid duration: %w", raw, err))
+
+	t, err := parseTime(raw)
+	if err == nil {
+		return t, nil
+	}
+	errs = append(errs, fmt.Errorf("%s is not a valid time: %w", raw, err))
+
+	return nil, errors.Join(errs...)
+}
+
+var (
+	supportedLayouts = []string{
+		time.RFC3339,
+		metav1.RFC3339Micro,
```

Comment:
Does one of these two layouts also include the Unix timestamp? If not, then I don't know if/where that case is parsed. I'm also curious to know if it wants seconds, milliseconds, microseconds, etc.

Reply:
Nice catch @g-linville! Parsing was missing for unix timestamps.

I went ahead and added that. I also added some comments (w/ examples) to the `parseTimeBound` function to clarify the accepted formats.

LMK if that helps.